### ISSUE TITLE
BL helper behaviors overriden by app not available to SL

### ISSUE THREAD

**Issue by rwd** [sul-dlss/spotlight#1154]
Blacklight's strategy for allowing helpers to be overriden by an app, by using behavior modules, does not work with Spotlight.

For example, given a helper in the app containing:
```ruby
module UrlHelper
  include Blacklight::UrlHelperBehavior

  def url_for_document(doc, options = {})
    '/special/path'
  end
end
```

Blacklight will use the app's overriden helper method, but Spotlight will not.

I expect that this is due to the Rails engine having isolated namespace and so being unaware of overrides in the app.

**Comment by cbeer** [comment]
I think we got around that for some other helpers by whitelisting the relevant application helpers:

```ruby
   def main_app_url_helper?(method)
      (method.to_s.end_with?('_path') || method.to_s.end_with?('_url')) &&
        main_app.respond_to?(method)
    end
```
https://github.com/sul-dlss/spotlight/blob/master/app/helpers/spotlight/application_helper.rb#L156-159

I wonder if we should add that to the particular route to the whitelist, and hope there aren't too many similar ones?

Another couple workarounds I can think of are::

- push `url_for_document` into a `Blacklight::DocumentPresenter`
- change the application's `config/routes.rb` file to do the appropriate routing.
- use `blacklight_config.show.route` [1] instead of overriding the helper
- use Rails' polymorphic routing and implement `#to_model` or other appropriate Rails model routing methods

@jcoyne @jkeck @mejackreed ?

[1] https://github.com/projectblacklight/blacklight/blob/master/app/helpers/blacklight/url_helper_behavior.rb#L13-18

**Comment by rwd** [comment]
I don't think that the whitelist in `Spotlight::ApplicationHelper#main_app_url_helper?` will help here because the method is not missing; it is provided by Blacklight and so `#main_app_url_helper?` never gets called for it.
